packages/compiler/src/optimizer/optimize.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  componentList,
+  defaultComponentTag,
+} from "@kuma-ui/core/components/componentList";
+import {
+  JsxOpeningElement,
+  JsxSelfClosingElement,
+  Node,
+  SyntaxKind,
+} from "ts-morph";
+import { isStyledProp, isPseudoProps } from "@kuma-ui/system";
+import { decode } from "../collector/decode";
+import { handleJsxExpression } from "../collector/expression";
+
+export const optimize = (
+  componentName: (typeof componentList)[keyof typeof componentList],
+  jsxElement: JsxOpeningElement | JsxSelfClosingElement,
+  as?: string
+) => {
+  const isOptimizable = jsxElement.getAttributes().every((attrLike) => {
+    if (Node.isJsxSpreadAttribute(attrLike)) return false;
+    const attr = attrLike.asKindOrThrow(SyntaxKind.JsxAttribute);
+    if (hasDynamicProp(attr.getNameNode().getText())) return false;
+    return true;
+  });
+
+  if (!isOptimizable) return;
+
+  const rawHTMLTag = (() => {
+    const tag = defaultComponentTag[componentName];
+    if (!!as && typeof as === "string") {
+      jsxElement.getAttribute("as")?.remove();
+      return (as as string).replace(/['"`]/g, "");
```

Comment:
I think it's better to make sure `as` does not include things like `'` at this point rather than adding special handling here, which is error prone

Reply:
@naruaway 
Still a bit ad-hoc but fixed it [6375b38](https://github.com/kuma-ui/kuma-ui/pull/240/commits/6375b38f84d375d118604f52d342a1550fd2033b)